tests/subscriptions/entity_subscriptions/test_entity_subscriptions.py

```diff
@@ -1,158 +1,296 @@
+#         True,
+#         id="Sessions subscription of type SNQL",
+#     ),
+#     pytest.param(
+#         MetricsCountersSubscription(data_dict={"organization": 1}),
+#         [
+#             binary_condition(
+#                 ConditionFunctions.EQ,
+#                 Column(None, None, "org_id"),
+#                 Literal(None, 1),
+#             ),
+#         ],
+#         True,
+#         id="Metrics counters subscription of type SNQL",
+#     ),
+#     pytest.param(
+#         MetricsSetsSubscription(data_dict={"organization": 1}),
+#         [
+#             binary_condition(
+#                 ConditionFunctions.EQ,
+#                 Column(None, None, "org_id"),
+#                 Literal(None, 1),
+#             ),
+#         ],
+#         True,
+#         id="Metrics sets subscription of type SNQL",
+#     ),
+# ]
+
 
-@pytest.mark.parametrize("entity_subscription, creation_dict, exception", TESTS)
-def test_basic(
-    entity_subscription: Type[EntitySubscription],
-    creation_dict: Mapping[str, Any],
+@pytest.mark.parametrize("entity_key, query, metadata, exception", TESTS)
+def test_entity_subscription_processors(
+    entity_key: EntityKey,
+    query: Union[CompositeQuery[QueryEntity], Query],
+    metadata: Mapping[str, Any],
     exception: Optional[Type[Exception]],
 ) -> None:
-    if exception is not None:
-        with pytest.raises(exception):
-            entity_subscription(**creation_dict)
-    else:
-        es = entity_subscription(**creation_dict)
-        assert es.to_dict() == creation_dict["data_dict"]
+    entity_subscription = get_entity(entity_key).get_entity_subscription()
 
+    if entity_subscription and entity_subscription.processors:
+        for processor in entity_subscription.processors:
+            if exception is not None:
+                with pytest.raises(exception):
+                    processor.to_dict(metadata) == {}
+            else:
+                if isinstance(processor, AddColumnCondition):
+                    processor.process(query, metadata)
+                    new_condition = query.get_condition()
+                    assert isinstance(new_condition, FunctionCall)
+                    assert len(new_condition.parameters) == 2
+                    assert processor.to_dict(metadata) == {
+                        processor.extra_condition_data_key: 1
+                    }
 
-@pytest.mark.parametrize(
-    "entity_subscription, expected_conditions, has_offset", TESTS_CONDITIONS_SNQL_METHOD
-)
-def test_entity_subscription_methods_for_snql(
-    entity_subscription: EntitySubscription,
-    expected_conditions: List[Any],
-    has_offset: bool,
+
+@pytest.mark.parametrize("entity_key, query, metadata, exception", TESTS)
+def test_entity_subscription_validators(
+    entity_key: EntityKey,
+    query: Union[CompositeQuery[QueryEntity], Query],
+    metadata: Mapping[str, Any],
+    exception: Optional[Type[Exception]],
 ) -> None:
-    offset = 5 if has_offset else None
```

Comment:
This was testing the offset conditions being correctly added. Are there still tests for that somewhere?

Reply:
The previous tests didn't actually check anything because offset was not used. So all tests with or without offset would just resolve which is why I removed it. Added it back in here in case we use/test offset in the future.